### PR TITLE
Allow to specify TS target using parserOptions

### PR DESCRIPTION
**Pull request by ptrxyz**
Suggestion for https://github.com/ota-meshi/eslint-plugin-svelte/issues/374

### CODE REVIEW

File: src/shared/svelte-compile-warns/transform/typescript.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export function transform(
     const output = ts.transpileModule(code, {
       reportDiagnostics: false,
       compilerOptions: {
-        target: ts.ScriptTarget.ESNext,
+        target: context.parserServices.program.getCompilerOptions().target || ts.ScriptTarget.ESNext,
```
Comment (ota-meshi) [inline src/shared/svelte-compile-warns/transform/typescript.ts:32]: Thank you for this PR.
Just to be sure, could you please change the following to use optional chaining?

```suggestion
        target: context.parserServices.program?.getCompilerOptions()?.target || ts.ScriptTarget.ESNext,
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: src/shared/svelte-compile-warns/transform/typescript.ts (hunk 1 above)
Comment (ptrxyz) [reply src/shared/svelte-compile-warns/transform/typescript.ts]: Yep, done.

File: src/shared/svelte-compile-warns/transform/typescript.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export function transform(
     const output = ts.transpileModule(code, {
       reportDiagnostics: false,
       compilerOptions: {
-        target: ts.ScriptTarget.ESNext,
+        target: context.parserServices.program?.getCompilerOptions()?.target || ts.ScriptTarget.ESNext,
```
Comment (ota-meshi) [inline src/shared/svelte-compile-warns/transform/typescript.ts:32]: It doesn't seem to work for our test cases. I think we probably need to add an option like this:

```suggestion
        target: context.parserServices.program?.getCompilerOptions()?.target || ts.ScriptTarget.ESNext,
        module: ts.ModuleKind.ESNext,
```

Also, Could you format it using `yarn lint-fix`?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Could you format it using yarn lint-fix?

**Review by ota-meshi — APPROVED** [review]
LGTM!

**Pull request merged**